clients/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BaseUseReferenceExec.scala

```diff
@@ -34,7 +34,12 @@ abstract class BaseUseReferenceExec(
   ): Seq[InternalRow] = {
 
     val ref = NessieUtils.calculateRef(branch, timestampOrHash, api)
-    NessieUtils.setCurrentRefForSpark(currentCatalog, catalog, ref)
+    NessieUtils.setCurrentRefForSpark(
+      currentCatalog,
+      catalog,
+      ref,
+      null != NessieUtils.hashFrom(timestampOrHash)
```

Comment:
If it is a `timestamp`, `NessieUtils.hashFrom` will return null, but we what we want to set is proper hash mapping to that timestamp.